### PR TITLE
Follow-up for uuidd/lastlog2 tmpfiles/service

### PR DESCRIPTION
**Pull request by YHNdnzj**

### CODE REVIEW

File: misc-utils/meson.build [hunk 1]
```diff
@@ -76,15 +76,17 @@ test_uuidd_sources = files(
   'test_uuidd.c',
 )
 
-if build_liblastlog2 and systemd.found()
```
Comment (karelzak) [inline misc-utils/meson.build:79]: In this case, we also need a default for $tmpfilesdir in ./configure.ac. Now it depends on `pkg-config systemd`.

File: misc-utils/uuidd.service.in [hunk 2]
```diff
@@ -16,7 +16,7 @@ ProtectKernelTunables=yes
 ProtectKernelModules=yes
 ProtectControlGroups=yes
 MemoryDenyWriteExecute=yes
-ReadWritePaths=/var/lib/libuuid/
+StateDirectory=libuuid
```
Comment (karelzak) [inline misc-utils/uuidd.service.in:19]: I wouldn't say I like this change. It makes things fragile as the rest of the code assumes "/var/lib/libuuid". 

We need to change the path in the service file to @localstatedir@/libuuid. 

Note that the path is hardcoded in libuuid. Using the same path everywhere would be more robust than only in service file follow any 3rd-party shorcut.  Imagine that I'll compile libuuid/uuidd with --localstatedir=/foo/bar than the file will be useless.

Comment (karelzak) [reply misc-utils/uuidd.service.in:19]: I'll fix the hardcoded path in the service file (we need @localstatedir@/libuuid there).

Comment (YHNdnzj) [reply misc-utils/uuidd.service.in:19]: Honestly, I feel this case quite synthetic. If people don't follow normal prefix, then they are unlikely to use the service either... But OK.

File: misc-utils/meson.build (hunk 1 above)
Comment (YHNdnzj) [reply misc-utils/meson.build:79]: I've never written autotools config before and I have no idea how to correctly do this... My main intention with this PR is to switch to `StateDirectory=`. Now that it is refused I'm tempted to close this.

File: misc-utils/uuidd.service.in (hunk 2 above)
Comment (karelzak) [reply misc-utils/uuidd.service.in:19]: This is the difference between upstream and downstream. It's fine to assume anything in your distro and prepare a package for the distro, but for upstream it's difficult to assume anything and at the same time support a large range different usecases (and paths) and configurations.

**Comment by karelzak** [comment]
Not merged, but thanks for your feedback!

**Pull request closed without merging**